index.rst

```diff
@@ -1,61 +1,263 @@
+If one component becomes unavailable, the others are expected to continue to operate, taking precautious to deal with the missing agent.
+At the same time, it can become quite burdensome to update and grow these systems as their complexity increases exponentially with the number of different components.
 
-  Subsubsections
-  ^^^^^^^^^^^^^^
+In any case, the middleware plays a crucial part in any kind of *distributed system*, acting as the glue that binds the system together.
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+The Vera Rubin Observatory Control System (Rubin-OCS) is designed following an *event-driven*, *reactive*, *distributed* architecture.
+The system is composed of a number of independent components that work together to execute cohesive operations.
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+The middleware is encapsulated with a layer of abstraction known as the Service Abstraction Layer (SAL), which uses the ADLink-OpenSpliceDDS implementation of the Data Distribution Service (DDS) message passing system.
 
-     Caption text.
+With this high-level overview of the system in mind, we will now focus on SAL and the different aspects of the current middleware technology (ADLink-OpenSpliceDDS).
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+The Past
+========
 
-   Feel free to delete this instructional comment.
+Probably the most important thing for us to consider when speaking about the past, is trying to understand why DDS was selection in the first place and then why the ADLink-OpenSpliceDDS implementation was adopted.
 
-:tocdepth: 1
+To put it in perspective, the firt commit to the current `SAL code repository`_ dates back to August 2014, some 8 years from the time of this writting and almost a whole year before the construction `first stone`_.
+For comparisom, the Apache Kafka message system first stable release dates back to January 2011, whereas the `DDS version 1.0 standard` dates back to December 2004.
+
+.. _SAL code repository: https://github.com/lsst-ts/ts_sal
+.. _first stone: https://www.nsf.gov/news/news_summ.jsp?cntn_id=134805&org=NSF&from=news
+.. _DDS version 1.0 standard: https://www.omg.org/spec/DDS/1.0
+
+At the time when decision was being made about selecting a middleware technology, DDS was a mature standard.
+The technology defines a powerfull real-time message system protocol with high inter-operability between platforms and programming languanges.
+In fact, DDS has most of the important features we recognize as crucial for the Rubin-OCS.
+Some of the most important ones are, for example:
```

Comment:
Consider replacing this sentence with a short extension to the previous, e.g.: "crucial for the for the Rubin-OCS, including:"